histogram/histogram.go

```diff
@@ -0,0 +1,54 @@
+// Package histogram provides primitives for working with histograms,
+// particularly towards Prometheus exporters.
+package histogram
+
+import (
+	"github.com/giantswarm/microerror"
+)
+
+type Config struct {
+	// BucketLimits is the upper limit of each bucket in the histogram.
+	// See https://godoc.org/github.com/prometheus/client_golang/prometheus#HistogramOpts.
+	BucketLimits []float64
+}
+
+// Histogram is a data structure suitable for providing the inputs to a Prometheus Histogram.
+// See https://godoc.org/github.com/prometheus/client_golang/prometheus#MustNewConstHistogram.
+type Histogram struct {
+	// Count is the number of entries added to the Histogram.
+	Count uint64
+	// Sum is the total sum of all entries added to the Histogram.
+	Sum float64
+	// Buckets is a map of upper bounds to cumulative counts of entries,
+	// excluding the +Inf bucket.
+	Buckets map[float64]uint64
```

Comment:
IMO these should be private fields. They can be configured with the config structure and when we have to marshal or get them out we can implement certain interfaces or getters or whatever.

Reply:
Let's go with the `Metric` method then, that was my thinking for having these private.